service/test/controllers/TelemetryControllerTest.java

```diff
@@ -0,0 +1,157 @@
+package controllers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.Props;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.controllers.BaseController;
+import play.mvc.Http.RequestBuilder;
+import play.mvc.Result;
+import play.test.FakeApplication;
+import play.test.Helpers;
+import util.Constant;
+import util.Message;
+
+/**
+ * This class will contains test cases for telemetry controller. this will mock the actor call.
+ *
+ * @author Mahesh Kumar Gangula
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(PowerMockRunner.class)
+public class TelemetryControllerTest {
```

Comment:
Missing scenarios:

1. request with no body or gzip files
2. request with empty string as body
3. request with empty gzip file

Reply:
added 1 and 2.

Reply:
what about the 3rd scenario?